### ISSUE TITLE
SelectionManager changes

### ISSUE THREAD

**Issue by justinvdk** [ProgrammingLife2016/PL2-2016#203]
We now have a SelectionManager class. I think this class is getting too many responsibilities and we should introduce a new class, e.g. SelectionPaneController with a corresponding FXML file. We can than drop the fxml into the layout somewhere and set it up in the rootLayoutController.java.

Then a refactor is needed to eventually remove SelectionManager.
Things to consider:
- #202
- Where do we now keep track of the top/bottom Genomes)
- Can we only select a single thing at a time, or can we select 1 thing in the tree + 1 thing in the graph etc.
- Consistent naming of FXML's
- Probably more

**Comment by justinvdk** [comment]
New plan:
SelectionManager stays, but in a much more lighter form.
Selection isn't that easy as it sounds, there are different types and usually the actor in selecting (a visual node) does not share a similar lifecycle with its data object.  

GraphPaneController and TreePaneController are now responsible for their own selection.
This might seem odd, please do share your opinion but this is the challenge:

A lot of "actors" in selection, i.e. the JavaFX Node's, do not have a similar lifecycles with their corresponding data nodes. And thus there is no good way to reference back to have the SelectionManager KNOW whether a given (JavaFX) Node is selected. There ARE ways, but since they differ for graphnodes and treenodes it gets complicated.

For now this is my (hacky) solution:
ISelectionInfo now includes a 'getSelectionIdentifier' method which returns an Object. The idea is that the object will be used to check for equality. The graphNodes return their corresponding datanodes, and the treeNodes theirs. This way the JavaFX panes can be redrawn many times without losing the information about what is selected, and a way to highlight these JavaFX Nodes when one appears to correspond to a selected item.

**Comment by justinvdk** [comment]
Furthermore, the new SelectionManager knows 'categories', for now only Graph and Tree (and I think it will probably stay this way). For each category the user is able to multi-select items.

**Comment by justinvdk** [comment]
See related commit. @Pathemeous please don't hate :(

**Comment by justinvdk** [comment]
```
              +--->selectedGraphNodes
              +--->selectedTreeNodes
              |
          +----------------+
     +--->+SelectionManager+<---+
     |    +----------------+    |
     |           ^              |
     v           |              v
  TreePane       |         SelectionPane
  Controller     v         Controller
              GraphPane
              Controller
```

So every component makes sure it notifies the SelectionManager when something is selected/deselected. The selectionManager keeps track. Every component is then allowed to listen for changes and update their view accordingly.

**Comment by justinvdk** [comment]
closed with #213
